app/uk/gov/hmrc/tariffclassificationfrontend/controllers/ActionsFactory.scala

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.tariffclassificationfrontend.controllers
+
+import javax.inject.{Inject, Singleton}
+import play.api.mvc.Results._
+import play.api.mvc.{ActionFilter, ActionRefiner, Result}
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.HeaderCarrierConverter
+import uk.gov.hmrc.tariffclassificationfrontend.models.Case
+import uk.gov.hmrc.tariffclassificationfrontend.models.request.AccessType._
+import uk.gov.hmrc.tariffclassificationfrontend.models.request.{AuthenticatedCaseRequest, AuthenticatedRequest}
+import uk.gov.hmrc.tariffclassificationfrontend.service.CasesService
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+import scala.concurrent.Future.successful
+
+@Singleton
+class CheckPermissionsAction
+  extends ActionRefiner[AuthenticatedCaseRequest, AuthenticatedCaseRequest] {
+
+  override protected def refine[A](request: AuthenticatedCaseRequest[A]):
+  Future[Either[Result, AuthenticatedCaseRequest[A]]] = {
+
+    if (request.c.isAssignedTo(request.operator)) {
```

Comment:
can we call the case something better than `c` please. You can use backticks and call it ``case``